File: 0268-missing-number/0268-missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n = nums.size();
        int totalsum = (n * (n+1))/2;
        int sum = accumulate(nums.begin(), nums.end(), 0);
        return totalsum - sum;
    }
};

File: 0268-missing-number/0268-missing-number.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        xor_actual = 0
        xor_curr = 0

        for i in range(len(nums) + 1):
            if i < len(nums):
                xor_curr ^= nums[i]
            xor_actual ^= i
        
        return xor_curr ^ xor_actual
